plugins/chat/app/controllers/chat/api/direct_messages_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# TODO (martin) Remove this endpoint when we move to do the channel creation
+# when a message is first sent to avoid double-request round trips for DMs.
+class Chat::Api::DirectMessagesController < Chat::ApiController
+  def create
+    with_service(Chat::CreateDirectMessageChannel) do
+      on_success do
+        render_serialized(
+          result.channel,
+          Chat::ChannelSerializer,
+          root: "channel",
+          membership: result.membership,
+        )
+      end
+      on_model_not_found(:target_users) { raise ActiveRecord::RecordNotFound }
+      on_failed_policy(:does_not_exceed_max_direct_message_users) do |policy|
+        raise Discourse::InvalidParameters.new(:target_usernames, policy.reason)
+      end
+      on_failed_policy(:acting_user_not_disallowing_all_messages) do
+        render_json_error(I18n.t("chat.errors.actor_disallowed_dms"))
+      end
+      on_failed_policy(:acting_user_can_communicate_with_all_parties) do |policy|
+        render_json_error(policy.reason)
+      end
```

Comment:
I think we try to be too verbose here.

I suggest the following:

```
on_failed_policy(:satisfies_max_direct_message_users)
on_failed_policy(:targets_allow_dms)
on_failed_policy(:targets_allow_dms_from_user)
```

Reply:
These make sense except the second one, it should be more like `actor_allows_dms`. Will change :+1: